geopandas/geodataframe.py

```diff
@@ -1738,7 +1753,7 @@ def astype(self, dtype, copy=True, errors="raise", **kwargs):
         -------
         GeoDataFrame or DataFrame
         """
-        df = super().astype(dtype, copy=copy, errors=errors, **kwargs)
+        df = pd.DataFrame(self).astype(dtype, copy=copy, errors=errors, **kwargs)
```

Comment:
Due to the try/except change above, astype(object) gets promoted to geometrydtype. 
I've changed this in a diff minimising way, but it would be better to downcast this case explicitly and simplify the method rather than downcast all cases and then upcast the gdf cases.

(A bit annoying, until this change this PR would have meant we could completely remove the astype overload I think.)